adapters/socketio_mbed.c

```diff
@@ -223,7 +226,11 @@ int socketio_open(CONCRETE_IO_HANDLE socket_io, ON_IO_OPEN_COMPLETE on_io_open_c
     }
     else
     {
-        socket_io_instance->tcp_socket_connection = tcpsocketconnection_create();
+        if (socket_io_instance->tcp_socket_connection == NULL)
```

Comment:
This could lead to a memory leak if open is called more than once. The previous code was already like that. We should fix it in this change.